lane_planner/nodes/lane_select/lane_select_core.cpp

```diff
@@ -839,12 +834,20 @@ int32_t getClosestWaypointNumber(const autoware_msgs::Lane &current_lane, const
   dist_vec.reserve(idx_vec.size());
   for (const auto &el : idx_vec)
   {
-    double dt = getTwoDimensionalDistance(current_pose.position, current_lane.waypoints.at(el).pose.pose.position);
-    dist_vec.push_back(dt);
+    double distance = getTwoDimensionalDistance(
+      current_pose.position, current_lane.waypoints.at(el).pose.pose.position);
+    dist_vec.push_back(distance);
   }
+
+  // Check distance
   std::vector<double>::iterator itr = std::min_element(dist_vec.begin(), dist_vec.end());
-  int32_t found_number = idx_vec.at(static_cast<uint32_t>(std::distance(dist_vec.begin(), itr)));
-  return found_number;
+  if (*itr > distance_threshold)
+  {
+    return -1;
```

Comment:
I think it makes sense to maintain the `ROS_WARN` that was removed above to let the user know why a solution was not found.

Reply:
It was removed since there are already some similar ROS_WARN messages on line 194 and 353 that use the output of this function.
Would changing to ROS_DEBUG be better maybe?

Reply:
No, those are fine. Thanks for clarification!